src/main/java/duke/Parser.java

```diff
@@ -0,0 +1,181 @@
+package duke;
+
+import duke.command.*;
+import duke.exception.DukeException;
+
+import java.time.LocalDate;
+import java.time.LocalDateTime;
+import java.time.format.DateTimeFormatter;
+import java.time.format.DateTimeParseException;
+
+/**
+ * A class to process and make sense of the command
+ * inputted by the user.
+ */
+
+public class Parser {
+
+    private Ui ui;
+    private TaskList taskList;
+
+    /**
+     * A public constructor for Parser to initialize the
+     * Ui and TaskList to the given ones.
+     *
+     * @param ui       The Ui to deal with interactions of the user.
+     * @param taskList the list storing all the tasks.
+     */
+    public Parser(Ui ui, TaskList taskList) {
+
+        this.ui = ui;
+        this.taskList = taskList;
+    }
+
+    /**
+     * Returns a Command representing the command
+     * inputted by the user.
+     *
+     * @param command The user command.
+     * @return A Command representing the user command.
+     * @throws DukeException If the command is inputted wrongly.
+     */
+    public Command parse(String command) throws DukeException {
+        String[] splitInput = command.trim().split(" +");
+        switch (splitInput[0]) {
+
+        case "bye":
+            return new ExitCommand(command);
+
+
+        case "list":
+            if (splitInput.length > 1) {
+                throw new DukeException(ui.commandError());
+            }
+            return new ListCommand(command);
+
+        case "find":
+            if (splitInput.length == 1) {
+                throw new DukeException(ui.commandError());
+            }
+            return new FindCommand(command);
+
+        case "done":
+            return checkDone(command);
+
+        case "delete":
+            return checkDelete(command);
+
+        case "todo":
+        case "event":
+        case "deadline":
```

Comment:
You can try to mark `// fall through` in these 2 lines to make it notable.

Reply:
Agreed.